addons/web/static/tests/views/fields/reference_field_tests.js

```diff
@@ -525,6 +526,44 @@ QUnit.module("Fields", (hooks) => {
         );
     });
 
+    QUnit.test("Many2One 'Search More...' updates on resModel change", async function (assert) {
+
+        // Patch the Many2XAutocomplete default search limit options
+        patchWithCleanup(Many2XAutocomplete.defaultProps, {
+            searchLimit: -1,
+        });
+
+        serverData.views = {
+            "product,false,list": '<tree><field name="display_name"/></tree>',
+            "product,false,search": '<search/>',
+        };
+
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: '<form><field name="reference"/></form>',
+        });
+
+        const refField = target.querySelector("div.o_field_reference");
```

Comment:
It's not a good idea to make a const and reuse it after an edition. The components could be rendered and your HTML element would no longer be the right one.